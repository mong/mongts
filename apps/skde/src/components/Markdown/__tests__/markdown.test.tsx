/**
 * To view a rendered version of all the snapshots:
 *
 * Copy the contents of ./__snapshots__/index.tsx.snap
 * Go to https://codesandbox.io/s/snapshot-viewer-lqjjsm?file=/src/snapshots.js
 * Replace the contents with what you copied
 */

import { render } from "@testing-library/react";
import { Markdown } from "..";

test("Standard render", async () => {
  const markdownText = "Dette er en test";
  const { container } = render(<Markdown>{markdownText}</Markdown>);
  expect(container).toMatchSnapshot();
});
test("Longer text", async () => {
  const markdownText =
    "Dette er en test\n\n# Hovedoverskrift\nLitt mer tekst\n## Underoverskrift\n### Dette er en overskrift!\n\n```sas\ndata test;\nset test2;\nvar = 0;\nrun;\n```\n\nOg litt tekst her";
  const { container } = render(<Markdown>{markdownText}</Markdown>);
  expect(container).toMatchSnapshot();
});

test("Text with image", async () => {
  const markdownText =
    'Dette er en test\n\n# Hovedoverskrift\n\n![Menisk, utvikling i antall inngrep i perioden 2013–2017](https://picsum.photos/id/237/100/100.jpg "Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.")\n\n';
  const { container } = render(<Markdown>{markdownText}</Markdown>);
  expect(container).toMatchSnapshot();
});

test("Text with image, language english", async () => {
  const markdownText =
    'Dette er en test\n\n# Hovedoverskrift\n\n![Menisk, utvikling i antall inngrep i perioden 2013–2017](https://picsum.photos/id/237/100/100.jpg "Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.")\n\n';
  const { container } = render(<Markdown lang="en">{markdownText}</Markdown>);
  expect(container).toMatchSnapshot();
});

test("Text with table", async () => {
  const markdownText =
    "Listen under viser hvilke helseforetak eller sykehus det er definert opptaksområder for og kortnavnene på disse som brukes i atlaset.\n\n| **Opptaksområde for**                 | **Kortnavn**    |\n| ------------------------------------- | --------------- |\n| Finnmarkssykehuset HF                 | Finnmark        |\n| Universitetssykehuset i Nord-Norge HF | UNN             |\n| Nordlandssykehuset HF                 | Nordland        |\n| Helgelandssykehuset HF                | Helgeland       |\n| Helse Nord-Trøndelag HF               | Nord-Trøndelag  |\n| St. Olavs hospital HF                 | St. Olavs       |\n| Helse Møre og Romsdal HF              | Møre og Romsdal |\n| Helse Førde HF                        | Førde           |\n| Helse Bergen HF                       | Bergen          |\n| Helse Fonna HF                        | Fonna           |\n| Helse Stavanger HF                    | Stavanger       |\n| Sykehuset Østfold HF                  | Østfold         |\n| Akershus universitetssykehus HF       | Akershus        |\n| Oslo universitetssykehus HF           | OUS             |\n| Lovisenberg diakonale sykehus         | Lovisenberg     |\n| Diakonhjemmet sykehus                 | Diakonhjemmet   |\n| Sykehuset Innlandet HF                | Innlandet       |\n| Vestre Viken HF                       | Vestre Viken    |\n| Sykehuset i Vestfold HF               | Vestfold        |\n| Sykehuset Telemark HF                 | Telemark        |\n| Sørlandet sykehus HF                  | Sørlandet       |";
  const { container } = render(<Markdown lang="en">{markdownText}</Markdown>);
  expect(container).toMatchSnapshot();
});

test("Text with details tag", async () => {
  const markdownText =
    "## Noe tekst\n<details><summary>Ledelse</summary>O-Store-Sol\n\n</details>";
  const { container } = render(<Markdown lang="en">{markdownText}</Markdown>);
  expect(container).toMatchSnapshot();
});

test("Text with url", async () => {
  const markdownText =
    "## Noe tekst\nTest [om](https://no.wikipedia.org) lenke fungerer\n";
  const { container } = render(<Markdown lang="en">{markdownText}</Markdown>);
  expect(container).toMatchSnapshot();
});
