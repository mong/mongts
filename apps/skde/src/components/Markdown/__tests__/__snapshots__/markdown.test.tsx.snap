// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Longer text 1`] = `
<div>
  <p>
    Dette er en test
  </p>
  

  <h1
    id="hovedoverskrift"
  >
    Hovedoverskrift
  </h1>
  

  <p>
    Litt mer tekst
  </p>
  

  <h2
    id="underoverskrift"
  >
    Underoverskrift
  </h2>
  

  <h3
    id="dette-er-en-overskrift"
  >
    Dette er en overskrift!
  </h3>
  

  <pre>
    <code
      class="language-sas"
    >
      data test;
set test2;
var = 0;
run;

    </code>
  </pre>
  

  <p>
    Og litt tekst her
  </p>
</div>
`;

exports[`Standard render 1`] = `
<div>
  <p>
    Dette er en test
  </p>
</div>
`;

exports[`Text with details tag 1`] = `
<div>
  <h2
    id="noe-tekst"
  >
    Noe tekst
  </h2>
  

  <details>
    <summary>
      Ledelse
    </summary>
    O-Store-Sol

  </details>
</div>
`;

exports[`Text with image 1`] = `
<div>
  <p>
    Dette er en test
  </p>
  

  <h1
    id="hovedoverskrift"
  >
    Hovedoverskrift
  </h1>
  

  <figure
    style="width: 90%; display: flex; justify-content: center; align-items: center; flex-direction: column;"
  >
    <img
      alt="Menisk, utvikling i antall inngrep i perioden 2013–2017"
      src="https://picsum.photos/id/237/100/100.jpg"
      style="display: block; margin: auto;"
      title="Menisk, utvikling i antall inngrep i perioden 2013–2017"
    />
    <figcaption>
      <strong>
        Figur:
      </strong>
       
      Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.
    </figcaption>
  </figure>
</div>
`;

exports[`Text with image without alt-text 1`] = `
<div>
  <p>
    Dette er en test
  </p>
  

  <h1
    id="hovedoverskrift"
  >
    Hovedoverskrift
  </h1>
  

  <figure
    style="width: 90%; display: flex; justify-content: center; align-items: center; flex-direction: column;"
  >
    <img
      alt="Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler."
      src="https://picsum.photos/id/237/100/100.jpg"
      style="display: block; margin: auto;"
      title=""
    />
    <figcaption>
      <strong>
        Figur:
      </strong>
       
      Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.
    </figcaption>
  </figure>
</div>
`;

exports[`Text with image, language english 1`] = `
<div>
  <p>
    Dette er en test
  </p>
  

  <h1
    id="hovedoverskrift"
  >
    Hovedoverskrift
  </h1>
  

  <figure
    style="width: 90%; display: flex; justify-content: center; align-items: center; flex-direction: column;"
  >
    <img
      alt="Menisk, utvikling i antall inngrep i perioden 2013–2017"
      src="https://picsum.photos/id/237/100/100.jpg"
      style="display: block; margin: auto;"
      title="Menisk, utvikling i antall inngrep i perioden 2013–2017"
    />
    <figcaption>
      <strong>
        Figure:
      </strong>
       
      Menisk, utvikling i antall inngrep pr. 100 000 innbyggere i perioden 2013–2017, justert for kjønn og alder. Fordelt på opptaksområder og offentlig eller privat behandler.
    </figcaption>
  </figure>
</div>
`;

exports[`Text with table 1`] = `
<div>
  <p>
    Listen under viser hvilke helseforetak eller sykehus det er definert opptaksområder for og kortnavnene på disse som brukes i atlaset.
  </p>
  






























































































  <table
    style="border-collapse: collapse; margin: auto;"
  >
    <thead>
      <tr>
        <th
          style="margin: 0px; padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          <strong>
            Opptaksområde for
          </strong>
        </th>
        <th
          style="margin: 0px; padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          <strong>
            Kortnavn
          </strong>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Finnmarkssykehuset HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Finnmark
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Universitetssykehuset i Nord-Norge HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          UNN
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Nordlandssykehuset HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Nordland
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Helgelandssykehuset HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Helgeland
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Helse Nord-Trøndelag HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Nord-Trøndelag
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          St. Olavs hospital HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          St. Olavs
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Helse Møre og Romsdal HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Møre og Romsdal
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Helse Førde HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Førde
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Helse Bergen HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Bergen
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Helse Fonna HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Fonna
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Helse Stavanger HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Stavanger
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Sykehuset Østfold HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Østfold
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Akershus universitetssykehus HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Akershus
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Oslo universitetssykehus HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          OUS
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Lovisenberg diakonale sykehus
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Lovisenberg
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Diakonhjemmet sykehus
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Diakonhjemmet
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Sykehuset Innlandet HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Innlandet
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Vestre Viken HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Vestre Viken
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Sykehuset i Vestfold HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Vestfold
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Sykehuset Telemark HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Telemark
        </td>
      </tr>
      <tr>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: left;"
        >
          Sørlandet sykehus HF
        </td>
        <td
          style="padding: calc(2rem / 16) calc(10rem / 16); text-align: right;"
        >
          Sørlandet
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Text with url 1`] = `
<div>
  <h2
    id="noe-tekst"
  >
    Noe tekst
  </h2>
  

  <p>
    Test 
    <a
      href="https://no.wikipedia.org"
      rel="noreferrer"
      target="_blank"
    >
      om
    </a>
     lenke fungerer
  </p>
</div>
`;
